metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/indexing_master.py

```diff
@@ -150,12 +151,36 @@ def zeppelin_notebook_import(self, env):
         env.set_params(params)
 
         Logger.info(ambari_format('Searching for Zeppelin Notebooks in {metron_config_zeppelin_path}'))
+
+        # With Ambari 2.5+, Zeppelin server is enabled to work with Shiro authentication, which requires user/password
+        # for authentication (see https://zeppelin.apache.org/docs/0.6.0/security/shiroauthentication.html for details).
+        ses = requests.session()
+
+        # Check if authentication is enabled on the Zeppelin server
+        try:
+            conn = ses.get(ambari_format('http://{zeppelin_server_url}/api/login'))
+
+            # Establish connection if authentication is enabled
+            try:
+                # The following credentials are created at install time by Ambari at /etc/zeppelin/conf/shiro.ini
+                # when Shiro auth is enabled on the Zeppelin server
+                zeppelin_payload = {'userName': 'admin', 'password' : 'admin'}
```

Comment:
Is this always going to be admin/admin or would it be the user/pass for the ambari instance?

Reply:
Hi @cestella , thanks for taking a look. The Zeppelin auth credentials are not related to ambari credentials. They are as defined in `/etc/zeppelin/conf/shiro.ini` on the Zeppelin host, which is set up by default during stack deploy time.

Here's a snippet:
```
[root@hcpca-1 ~]# cat /etc/zeppelin/conf/shiro.ini

[users]
# List of users with their password allowed to access Zeppelin.
# To use a different strategy (LDAP / Database / ...) check the shiro doc at http://shiro.apache.org/configuration.html#Configuration-INISections
admin = admin, admin
<snip>
```

When I created the pull request, I considered the option of reading `shiro.ini` file and parsing for the credentials. I did not pursue that further since I wanted to address the immediate need of ability to import notebook on an out-of-the-box Metron cluster. I understand that a thorough way would be to handle the [different INI settings](http://shiro.apache.org/configuration.html#Configuration-INISections) provided by shiro, and could potentially be the scope of another pull request. 

Please let me know if it makes sense.

Reply:
so, I wonder if we should add a configuration parameter for zeppelin user/pass.  If people change the default user/pass of zeppelin, this won't work, I think.  The other option is whether or not there is an existing config in ambari where this is held (maybe not though).  Beyond that, we could always parse the shiro.ini (even as simple as a little python snippet like:
```
with open('/etc/zeppelin/conf/shiro.ini', 'r') as f:
  seen_users = False
  for line in f:
    if '[users]' in line:
      seen_users = True
    if seen_users and 'admin = ' in line:
      tokens = line.strip().split(',')
      username = tokens[0].split(' ')[-1].strip()
      password = tokens[1].strip()
      print "username: {}, password: {}".format(username, password)
      break
```

I am just a bit concerned at having this go in as it is given this assumption, but I could probably be talked into it by another committer.

Reply:
Why wouldn't we expose this through the ambari configuration?

Reply:
Hi @cestella and @ottobackwards - I found a way to get the Shiro INI content from Ambari and parse the admin credentials. Please have a look at my latest commit.